docs/guidelines/git-signed-commits.md

```diff
@@ -0,0 +1,172 @@
+| **Cons**                   | - Primarily verifies the commit was pushed by an authenticated user, not necessarily the commit's author. <br> - SSH key management is crucial; compromised keys pose a risk. | - Key management and the web of trust model can be complex. <br> - Requires active key maintenance (revocation, expiration). | - Dependent on third-party CAs for issuance and trust. <br> - Certificates have expiration dates and may incur costs. |
+
+!!! Note Signing method
+
+    We recommend using self-signed SSH keys for signing your git commits (In the future we may switch to a certificate based approach)
+
+## Configure your local development environment
+
+For this guideline our reference is using Git from the command line. 
+
+We assume that Git and SSH is installed on your system. 
+
+We assume that you are using SSH to authenticate Git with github.com. Consult our [guideline](git-github.md) for more information on this topic.
+
+The Github documentation on [SSH commit signature verification](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification#ssh-commit-signature-verification) is a good source for detailed information.
+
+### Adding a SSH key for signing
+
+!!! Note Use multiple SSH keys
+
+    We recommend using different SSH keys for authentication and signing. This may add some extra complexity and it may provide a more robust set-up with looser coupling of key components of the SDLC.
+
+To create a new SSH key for signing you can use the following command (alter the date manually):
+
+```shell
+ssh-keygen -t ed25519 -f ~/.ssh/git_ssh_signing_key_1 -C "Created on <date>, for larskaare on github.com"
+```
+
+This will create a SSH signing key and add a comment on date and purpose. Add a passphrase to the key. Success full key generation will output the key fingerprint and a randomart image (randomart is supposed to be a visualisation making it easier to validate keys - and identify changes)
+
+Add the new key to the ssh-agent
+
+```shell
+ssh-add ~/.ssh/git_ssh_signing_key_1
+```
+
+### [Configure Git](https://docs.github.com/en/authentication/managing-commit-signature-verification/telling-git-about-your-signing-key#telling-git-about-your-ssh-key) to use the SSH key
+
+We will configure the git global settings to use the new SSH key for signing commits locally. Examples assume you created the key as defined above.
+
+```shell
+git config --global gpg.format ssh
+git config --global user.signingkey $HOME/.ssh/git_ssh_signing_key_1.pub
```

Comment:
Should we use `~` instead of `$HOME`, to be consistent with elsewhere in the document?

Reply:
Yes, i'll change to ~. ~ is a shell specific feature while $HOME is not. The behaviour can vary between systems so $HOME would most likle be the more precise and perhaps more reliable.